[PPM-10] Replace if_indextoname to network_utils::linux_get_iface_name

VladyslavTupikin opened the pull request:
Need to replace `if_indextoname()` to the wrapper `network_utils::linux_get_iface_name`. For all prplmesh source code.

https://jira.prplfoundation.org/browse/PPM-10

Commit:
agent: backhaul_manager: Replace if_indextoname()

Need to replace if_indextoname() to the wrapper
network_utils::linux_get_iface_name.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
agent: platform_manager: Replace if_indextoname()

Need to replace if_indextoname() to the wrapper
network_utils::linux_get_iface_name.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bcl: Replace if_indextoname()

Need to replace if_indextoname() to the wrapper
network_utils::linux_get_iface_name.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
common: bwl: Replace if_indextoname()

Need to replace if_indextoname() to the wrapper
network_utils::linux_get_iface_name.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

vitalybu commented on agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp:
```diff
@@ -134,7 +134,9 @@ wan_monitor::ELinkState wan_monitor::process()
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), IFNAMSIZ - 1, name);
+            name[IFNAMSIZ - 1] = '\0';
```
Since you're terminating the string everywhere, I guess it's better to simply have this logic moved into the `network_utils::linux_get_iface_name` method itself.

VladyslavTupikin replied:
@vitalybu `network_utils::linux_get_iface_name()` returns the C++ string. I think it is a bad idea to add '\0' to C++ string container.

vitalybu replied:
True, but first it uses a local variable (i.e. `iface_name`) for storing the output of `if_indextoname`.
You can terminate `iface_name` before returning it as `std::string`...

VladyslavTupikin replied:
Agree with you, will fix it!

VladyslavTupikin replied:
@vitalybu fixed!

Commit:
beerocks: network_utils: Add 0 terminator in linux_get_iface_name

Method linux_get_iface_name() operates C-style
strings and returns result as C++ string container.
In some places when we need to use this method
need to put result in C-style string and
it can cause a problem when we can get
the string without zero terminator.

Add '\0' as last character for iface_name
char array in the linux_get_iface_name()
method.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

vitalybu commented on agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp:
```diff
@@ -134,7 +134,8 @@ wan_monitor::ELinkState wan_monitor::process()
             // Convert the interface index to name
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), IFNAMSIZ, name);
```
This can access unallocated memory when the interface name is smaller than IFNAMSIZ:
```suggestion
            std::copy_n(iface_index_name.begin(), iface_index_name.length(), name);
```

Relevant for all other commits as well.

VladyslavTupikin replied:
Fixed

morantr commented on agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp:
```diff
@@ -134,7 +134,8 @@ wan_monitor::ELinkState wan_monitor::process()
             // Convert the interface index to name
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), iface_index_name.length(), name);
```
Why do you copy `iface_index_name` to `name`? You need to remove `name` completely.
Remove its declaration from line 133.
Remove the copy of `iface_index_name` to `name` on line 138.
Compare `m_strWanIfaceName` to `iface_index_name`  using `==` operator on line 137.
Print `iface_index_name` instead of `name` on line 146.

Also I would rename `iface_index_name` to `iface_name`.

VladyslavTupikin replied:
fixed

morantr commented on agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp:
```diff
@@ -1123,7 +1123,11 @@ bool main_thread::handle_arp_monitor()
+    if (!if_name_buffer[0]) {
         LOG(ERROR) << "Failed to find iface of iface_index" << int(iIfaceIndex);
         return false;
     }
```
Here also, remove `if_name_buffer` declaration, and put `linux_get_iface_name()` output directly on `strIfaceName`.

VladyslavTupikin replied:
fixed

morantr commented on common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:
```diff
@@ -1318,7 +1318,9 @@ bool mon_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
-        if_indextoname(nla_get_u32(tb[NL80211_ATTR_IFINDEX]), ifname);
+        auto index            = nla_get_u32(tb[NL80211_ATTR_IFINDEX]);
+        auto iface_index_name = beerocks::net::network_utils::linux_get_iface_name(index);
+        std::copy_n(iface_index_name.begin(), iface_index_name.length(), ifname);
```
Here also, declare `ifname` to be a `std::string` instead of `char[]`, and on each comparation to `ifname` use `==` operator, instead of `.compare()`.

VladyslavTupikin replied:
fixed
